src/reducers/CompareResults.ts

```diff
@@ -12,6 +13,7 @@ const comparisonResults = firstRevisionResults.concat(
 
 const initialState: CompareResultsState = {
   data: [],
+  results: [],
```

Comment:
### Optional

This is an optional suggestion.

- We can wait for all the calls to the API to finish before rendering the results table. 
To achieve this we can add a piece of state that tracks the number of API requests. 
For example `completedApiRequests`
```
const initialState: CompareResultsState = {
  data: [],
  completedApiRequests: 0,
  results: [],
  loading: false,
  error: undefined,
};
```
We can increment this number after each fetch 
```
  .addCase(fetchCompareResults.fulfilled, (state, action) => {
        state.data = action.payload;
        state.completedApiRequests += 1;
```

In ResultsTable component we can utilize `completedApiRequests` by conditionally rendering the processedResults in the table when `completedApiRequests === selectedRevisions.new.length` is true.

Reply:
Carla left some suggestions in the comments. I followed her approach instead so I'll resolve this conversation since it's marked as optional.